apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx

```diff
@@ -0,0 +1,550 @@
+  const zoomedInScale =
+    (ballotBounds.width / writeInBounds.width) * IMAGE_SCALE;
+
+  return (
+    <BallotImageViewerContainer>
+      <BallotImageViewerControls isZoomedIn={isZoomedIn}>
+        <Button onPress={() => setIsZoomedIn(false)} disabled={!isZoomedIn}>
+          <Icons.ZoomOut /> Zoom Out
+        </Button>
+        <Button onPress={() => setIsZoomedIn(true)} disabled={isZoomedIn}>
+          <Icons.ZoomIn /> Zoom In
+        </Button>
+      </BallotImageViewerControls>
+      {isZoomedIn ? (
+        <React.Fragment>
+          <WriteInFocusOverlay
+            focusWidth={writeInBounds.width * zoomedInScale}
+            focusHeight={writeInBounds.height * zoomedInScale}
+          />
+          <ZoomedInBallotImage
+            src={imageUrl}
+            alt="Ballot with write-in highlighted"
+            ballotBounds={ballotBounds}
+            writeInBounds={writeInBounds}
+            scale={zoomedInScale}
+          />
+        </React.Fragment>
+      ) : (
+        <ZoomedOutBallotImageContainer>
+          <img src={imageUrl} alt="Full ballot" />
+        </ZoomedOutBallotImageContainer>
+      )}
+    </BallotImageViewerContainer>
+  );
+}
+
+interface Props {
+  contest: CandidateContest;
+  election: Election;
+  onClose: () => void;
+}
+
+export function WriteInsAdjudicationScreen({
+  contest,
+  election,
+  onClose,
+}: Props): JSX.Element | null {
+  const queryClient = useQueryClient();
+  const apiClient = useApiClient();
+  const writeInsQuery = getWriteIns.useQuery({ contestId: contest.id });
+  const writeInCandidatesQuery = getWriteInCandidates.useQuery({
+    contestId: contest.id,
+  });
+  const adjudicateWriteInMutation = adjudicateWriteIn.useMutation();
+  const addWriteInCandidateMutation = addWriteInCandidate.useMutation();
+
+  const [offset, setOffset] = useState(0);
+  const currentWriteIn = writeInsQuery.data
+    ? writeInsQuery.data[offset]
+    : undefined;
+  const writeInImageViewQuery = getWriteInImageView.useQuery(
+    {
+      writeInId: currentWriteIn?.id ?? 'no-op',
+    },
+    !!currentWriteIn
+  );
+
+  const [showNewWriteInCandidateForm, setShowNewWriteInCandidateForm] =
+    useState(false);
+  const newWriteInCandidateInput = useRef<HTMLInputElement>(null);
+  const nextButton = useRef<Button>(null);
+  const firstAdjudicationButton = useRef<Button>(null);
+
+  // prefetch the next write-in image
+  useEffect(() => {
+    if (!writeInsQuery.data) return;
+    const nextWriteIn = writeInsQuery.data[offset + 1];
+    if (nextWriteIn) {
+      void queryClient.prefetchQuery({
+        queryKey: getWriteInImageView.queryKey({
+          writeInId: nextWriteIn.id,
+        }),
+        queryFn: () =>
+          getWriteInImageView.queryFn({ writeInId: nextWriteIn.id }, apiClient),
+      });
+    }
+  }, [apiClient, queryClient, writeInsQuery.data, offset]);
+
+  // as a modal, we can just return null
```

Comment:
The WIA screen is not really a "screen", it's a full-screen `modal`. so the user might just get a delay as those queries run before the `modal` pops up. My experience is that that delay is imperceptible for our current uses and scale. But the scaling question is a good one - if we need to fetch 1,000,000 write-ins rather than 20, maybe it gets very slow. I think the better approach there, however, is to later have a `limit` query so it'll still be fast and not be so much data flowing to the frontend.